File: LeetCode题目集/219. Contains Duplicate II(easy).java
import java.util.Hashtable;
import java.util.Map;

/*
 * @Author: SourDumplings
 * @Date: 2019-08-21 21:31:08
 * @Link: https://github.com/SourDumplings/
 * @Email: changzheng300@foxmail.com
 * @Description: https://leetcode.com/problems/contains-duplicate-ii/
 */

class Solution
{
    public boolean containsNearbyDuplicate(int[] nums, int k)
    {
        Map<Integer, Integer> map = new Hashtable<>();
        int n = nums.length;
        for (int i = 0; i < n; i++)
        {
            int num = nums[i];
            if (map.containsKey(num) && i - map.get(num) <= k)
            {
                return true;
            }
            map.put(num, i);
        }
        return false;
    }
}

File: OJ practices/LeetCode题目集/219. Contains Duplicate II(easy).cpp
/**
 * @file 219. Contains Duplicate II(easy).cpp
 * @author SourDumplings (changzheng300@foxmail.com)
 * @brief 
 * @version 1.0.0
 * @date 2024-05-02
 * 
 * @copyright Copyright (c) 2024 SourDumplings
 * 
 */

class Solution
{
public:
    bool containsNearbyDuplicate(vector<int> &nums, int k)
    {
        unordered_map<int, int> m;
        for (size_t i = 0; i < nums.size(); i++)
        {
            int num = nums[i];
            if (m.find(num) != m.end()
                && i - m.at(num) <= k
            )
            {
                return true;
            }
            m[num] = i;
        }
        return false;
    }
};